services/eventservice/src/pubsub_client.rs

```diff
@@ -0,0 +1,59 @@
+use super::types::{Category, Event};
+use cloud_pubsub::*;
+use std::env;
+use std::sync::Arc;
+
+#[derive(Debug)]
+struct UpdatePacket(String);
+
+impl FromPubSubMessage for UpdatePacket {
+    fn from(message: EncodedMessage) -> Result<Self, error::Error> {
+        match message.decode() {
+            Ok(bytes) => Ok(UpdatePacket(String::from_utf8_lossy(&bytes).into_owned())),
+            Err(e) => Err(error::Error::from(e)),
+        }
+    }
+}
+
+#[tokio::main]
+async fn test() {
```

Comment:
it seems to be test case.
Do you already match event data manager with certain topics?

Reply:
Pub sub layer is not started yet
And this test is just function name for testing if pubsub works at all